packages/alchemy/src/provider.ts

```diff
@@ -140,12 +139,6 @@ export class AlchemyProvider extends SmartAccountProvider<HttpTransport> {
       );
     }
 
-    if (this.feeOptsSet) {
-      return this.withPaymasterMiddleware(
-        alchemyPaymasterAndDataMiddleware(this, config)
-      );
-    } else {
-      return withAlchemyGasManager(this, config);
-    }
+    return withAlchemyGasManager(this, config, this.feeOptsSet);
```

Comment:
once we change the 3rd param to `estimateGas` this should probably be:
```suggestion
    return withAlchemyGasManager(this, config, !this.feeOptsSet);
```